packages/gatsby-remark-embed-snippet/src/index.js

```diff
@@ -27,23 +27,35 @@ const getLanguage = file => {
 }
 
 module.exports = (
-  { markdownAST },
+  { markdownAST, markdownNode },
   { classPrefix = `language-`, directory } = {}
 ) => {
   if (!directory) {
     throw Error(`Required option "directory" not specified`)
-  } else if (!fs.existsSync(directory)) {
-    throw Error(`Invalid directory specified "${directory}"`)
-  } else if (!directory.endsWith(`/`)) {
-    directory += `/`
+  }
+
+  let filePathResolve
+
+  if (typeof directory === `string`) {
+    if (!fs.existsSync(directory)) {
+      throw Error(`Invalid directory specified "${directory}"`)
+    }
+    if (!directory.endsWith(`/`)) {
+      directory += `/`
+    }
+
+    filePathResolve = file => normalizePath(`${directory}${file}`)
+  } else {
+    //its function
```

Comment:
It could be a lot of things besides a function actually. I'd tweak this!

```js
else if (typeof directory === `function`) {}
else {
 // warn?
}
```

Reply:
ok, fixed